src/ui/EditorNS/jsproxy.cpp

```diff
@@ -0,0 +1,116 @@
+#include "include/EditorNS/jsproxy.h"
+#include <QDebug>
+
+namespace EditorNS {
+
+QVariant JsToCppProxy::getResult()
+{
+    return m_result;
+}
+
+QVariant JsToCppProxy::getCursor()
+{
+    return m_cursor;
+}
+
```

Comment:
Could we do something like this?

bool JsProxy::value(&T, "cursor");

It would be a bool function that returned false on failure to set the variable data... or true if we successfully grabbed the information we needed.

This would still allow the conversion to happen within JsProxy, but it would be flexible in that it would accept any type as the first argument and get the data without too much hassle.